tox.ini

```diff
@@ -1,3 +1,193 @@
+[common]
+python_matrix = {36,37,38,39,310,311}
+postgres_matrix =
+    pg11: PG_MAJOR = 11
+    pg12: PG_MAJOR = 12
+    pg13: PG_MAJOR = 13
+    pg14: PG_MAJOR = 14
+    pg15: PG_MAJOR = 15
+psycopg_deps =
+    py{37,38,39,310,311}-{lin,win}: psycopg[binary]
+    mac: psycopg2-binary
+    py36: psycopg2-binary
+platforms =
+    lin: linux
+    mac: darwin
+    win: win32
+
+[tox]
+min_version = 4.0
+requires =
+    tox>4
+env_list =
+    dep
+    lint
+    py{[common]python_matrix}-test-{lin,mac,win}
+    docs
+skipsdist = True
+toxworkdir = {env:TOX_WORK_DIR:.tox}
+skip_missing_interpreters = True
+
+[testenv]
+setenv =
+    PYTHONDONTWRITEBYTECODE = 1
+    mac: OPEN_CMD = {env:OPEN_CMD:open}
+    lin: OPEN_CMD = {env:OPEN_CMD:xdg-open}
+passenv =
+    BROWSER
+    DISPLAY
+
+[testenv:lint]
+description = Lint code with flake8
+commands = flake8 {posargs:patroni tests setup.py}
+deps =
+    flake8
+
+[testenv:py{36,37,38,39,310,311}-test-{lin,win,mac}]
+description = Run unit tests with pytest
+labels =
+    test
+commands_pre =
+    - {tty:rm -f "{toxworkdir}{/}cov_report_{env_name}_html{/}index.html":true}
+    - {tty:rm -f "{toxworkdir}{/}pytest_report_{env_name}.html":true}
+commands =
+    pytest \
+    -p no:cacheprovider \
+    --verbose \
+    --doctest-modules \
+    --capture=fd \
+    --cov=patroni \
+    --cov-report=term-missing \
+    --cov-append \
+    {tty::--cov-report="xml\:{toxworkdir}{/}cov_report.{env_name}.xml"} \
+    {tty:--cov-report="html\:{toxworkdir}{/}cov_report_{env_name}_html":} \
+    {tty:--html="{toxworkdir}{/}pytest_report_{env_name}.html":} \
+    {posargs:tests patroni}
+commands_post =
+    - {tty:{env:OPEN_CMD} "{toxworkdir}{/}cov_report_{env_name}_html{/}index.html":true}
+    - {tty:{env:OPEN_CMD} "{toxworkdir}{/}pytest_report_{env_name}.html":true}
+deps =
+    -r requirements.txt
+    mock>=2.0.0
+    pytest
+    pytest-cov
+    pytest-html
+    {[common]psycopg_deps}
+platform =
+    {[common]platforms}
+allowlist_externals =
+    rm
+    {env:OPEN_CMD}
+
+[testenv:dep]
+description = Check package dependency problems
+commands = pipdeptree -w fail
+deps =
+    -r requirements.txt
+    pipdeptree
+    {[common]psycopg_deps}
+
+[testenv:py{37,38,39,310,311}-type-{lin,mac,win}]
+description = Run static type checking with pyright
+labels =
+    type
+deps =
+    -r requirements.txt
+    pyright
+    {[common]psycopg_deps}
+commands = pyright --venv-path {toxworkdir}{/}{envname} --lib {posargs:patroni}
```

Comment:
pyright complains about `--lib`:
```
The --lib option is deprecated. Pyright now defaults to using library code to infer types.
```

Reply:
Another problem pyright that we need to have both, psycopg2 and psycopg installed so it could resolve all dependencies.

Reply:
Fixed both of these